dns/r53/client.go

```diff
@@ -66,22 +66,34 @@ func (dns *dns) UpdateRecordSets(changes []*route53.Change) error {
 }
 
 // GetARecords gets a list of A Records from aws.
-// Todo tests
-func (dns *dns) GetARecords() ([]*route53.ResourceRecordSet, error) {
-	recordSetsOutput, err := dns.r53.ListResourceRecordSets(&route53.ListResourceRecordSetsInput{
+func (dns *client) GetARecords() ([]*route53.ResourceRecordSet, error) {
+	aRecords := []*route53.ResourceRecordSet{}
+	request := &route53.ListResourceRecordSetsInput{
 		HostedZoneId: aws.String(dns.hostedZone),
-	})
-
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch A records: %v", err)
 	}
+	for {
+		recordSetsOutput, err := dns.r53.ListResourceRecordSets(request)
 
-	recordSets := recordSetsOutput.ResourceRecordSets
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch A records: %v", err)
+		}
 
-	aRecords := []*route53.ResourceRecordSet{}
-	for _, recordSet := range recordSets {
-		if *recordSet.Type == "A" {
-			aRecords = append(aRecords, recordSet)
+		recordSets := recordSetsOutput.ResourceRecordSets
+
+		for _, recordSet := range recordSets {
+			if *recordSet.Type == "A" {
```

Comment:
`"A"`? I'm guessing aws library has a const for this we should probably use instead.